Formula/emscripten.rb

```diff
@@ -121,6 +134,9 @@ def install
     cd libexec do
       system "npm", "install", *Language::Node.local_npm_install_args
       rm_f "node_modules/ws/builderror.log" # Avoid references to Homebrew shims
+      # Delete native GraalVM image in incompatible platforms.
+      on_linux { rm_rf "node_modules/google-closure-compiler-linux" }
+      on_macos { rm_rf "node_modules/google-closure-compiler-osx" if Hardware::CPU.arm? }
     end
 
     %w[em++ em-config emar emcc emcmake emconfigure emlink.py emmake
```

Comment:
We probably also want to do something like this for the env scripts:
```diff
diff --git a/Formula/emscripten.rb b/Formula/emscripten.rb
index 5797ccf9c9..222ca3793a 100644
--- a/Formula/emscripten.rb
+++ b/Formula/emscripten.rb
@@ -123,9 +123,12 @@ class Emscripten < Formula
       rm_f "node_modules/ws/builderror.log" # Avoid references to Homebrew shims
     end
 
+    emscript_env = {PYTHON: Formula["python@3.9"].opt_bin/"python3"}
+    emscript_env[:JAVA_HOME] = Language::Java.overridable_java_home_env if Hardware::CPU.arm?
+
     %w[em++ em-config emar emcc emcmake emconfigure emlink.py emmake
        emranlib emrun emscons].each do |emscript|
-      (bin/emscript).write_env_script libexec/emscript, PYTHON: Formula["python@3.9"].opt_bin/"python3"
+      (bin/emscript).write_env_script libexec/emscript, emscript_env
     end
   end
```
We avoid assigning the env hash to `env` since the formula class defines an `env` method we should try to avoid shadowing.

Reply:
I think you forgot to assign `emscript_env`.

Reply:
Whoops!

Reply:
Do I need to define that in Linux too?

Reply:
Yes. You want `emscripten` executables to run with `PYTHON` set in the environment as pointing to Homebrew Python3.

Reply:
Sorry what I meant was whether we need to add `JAVA_HOME` on Linux like we do on ARM.

Reply:
Oh. I don't know, to be honest. I think it may not be needed -- I _think_ the reason we set it on macOS is that `openjdk` is keg-only. However, setting it doesn't hurt, so you could also do it.

Reply:
Okay I'll format it so it adds on ARM64 macOS or Linux.